python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -251,8 +252,16 @@ def assert_any_corrections(self, data):
         AssertionError
             If one of more attribute of the topic is set to True.
         """
+        corrections = []
+        for corr in self.valid_corrections:
+            if hasattr(data, corr):
+                corrections.append(corr)
+        if len(corrections) == 0:
+            raise IOError(f"No correction found in {data}. At least one of {self.valid_corrections} "
+                          f"must be present.")
+
         assert any([getattr(data, corr)
-                    for corr in self.valid_corrections]), \
+                    for corr in corrections]), \
```

Comment:
Regarding `def assert_corrections` below: I realize it is existing code but I suggest using a boolean instead of a string as an argument. (In fact you only call it once, with mode = "disable", but it's probably best to leave it flexible).

Also incrementally appending to a string is very inefficient:
```
        enabled = ''
        for key in self.corrections:
            if self.corrections[key]:
                enabled += key+','
```
It's better to make a list of strings and then concatenate them with `",".join(list)`. To make the list I suggest:
```
        enabled_keys = []
        for key, is_enabled in self.corrections.items():
            if is_enabled:
                enabled_keys.append(key)
```
Or if you don't mind a list comprehension:
```
        enabled_keys = [key for key, is_enabled in self.corrections.items() if is_enabled]
```
Finally, please don't raise IOError for an invalid argument; use ValueError or TypeError. I suggest looking for IOError everywhere in the code and replacing it.